File: Shared/Blog.Contracts/V1/Validators/UsersValidators/ChangePasswordRequestValidator.cs
using Blog.Contracts.V1.Requests.UsersRequests;
using FluentValidation;

namespace Blog.Contracts.V1.Validators.UsersValidators
{
    /// <summary>
    /// Change password request validator.
    /// </summary>
    /// <seealso cref="AbstractValidator{ChangePasswordRequest}" />
    public class ChangePasswordRequestValidator : AbstractValidator<ChangePasswordRequest>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ChangePasswordRequestValidator"/> class.
        /// </summary>
        public ChangePasswordRequestValidator()
        {
            RuleFor(x => x.OldPassword)
                .NotEmpty()
                .MinimumLength(6)
                .Matches("^[a-zA-Z0-9-_]*S");

            RuleFor(x => x.NewPassword)
                .NotEmpty()
                .MinimumLength(6)
                .Matches("^[a-zA-Z0-9-_]*S");
        }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/UsersValidators/ChangePasswordRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.UsersValidators

open FluentValidation
open Blog.FSharp.Contracts.V1.Requests.UsersRequests

/// <summary>
/// Validator for change password request.
/// </summary>
type ChangePasswordRequestValidator() =
    inherit AbstractValidator<ChangePasswordRequest>()

    do
        // Validate OldPassword: not empty, at least 6 chars, matches regex
        base.RuleFor(fun x -> x.OldPassword)
            .NotEmpty()
            .MinimumLength(6)
            .Matches("^[a-zA-Z0-9-_]*S") |> ignore

        // Validate NewPassword: not empty, at least 6 chars, matches regex
        base.RuleFor(fun x -> x.NewPassword)
            .NotEmpty()
            .MinimumLength(6)
            .Matches("^[a-zA-Z0-9-_]*S") |> ignore